### DISCUSSION TITLE
arduino 1.8.16 Attiny1607 board

### DISCUSSION THREAD

**Discussion by ebelouet** [SpenceKonde/megaTinyCore#542] [Q&A]
HI,

I try arduino 1.8.16 Attiny1607 board with Blink software and i add #define LED_BUILTIN 9 to change led pin, but I have nothing.

Do you know where is mistake ?

Sincerely
Eric

**Comment by phil-barrett** [comment]
Is the 1.8.15 bug still in 1.8.16?

**Reply by SpenceKonde** [reply]
What 1.8.15 bug?

**Reply by phil-barrett** [reply]
https://github.com/SpenceKonde/megaTinyCore/blob/master/Installation.md

**Reply by SpenceKonde** [reply]
Ah. updated that document. It does not impact my cores when they are installed through board manager. One of the dozen or so manuall or board manager installations visible to a non-portable copy blocks everything from compiling. I don't have time for that crap.

**Reply by phil-barrett** [reply]
OK, thanks.  good to know because Teensyduino seems to have issues with 1.8.16

**Reply by SpenceKonde** [reply]
I think it's back to the old "assume a new release is no good until proven otherwise" protocol. It's not like they've been putting anything useful in.

**Comment by SpenceKonde** [comment]
Nope...

```
#define LED_BUILTIN 9
void setup() {.
  pinMode(LED_BUILTIN, OUTPUT);
}

void loop() {
  digitalWrite(LED_BUILTIN, HIGH); 
  delay(1000);
  digitalWrite(LED_BUILTIN, LOW);
  delay(1000);
}
```
works for me. Have you by any chance expanded on blink further, particularly by turning on Serial?  pin 9 is PIN_PB2, the TX pin for serial, so if you've called Serial.begin(), you can't manually control the pin - you've handed that pin to the serial port until you call Serial.end() or reset it. Other than that, i'd suspect a wiring problem. 

Or a soldering problem, that VQFN package is nearly impossible to solder with an iron (frankly, it's no piece of cake to solder with a reflow oven either, I've had best results via one of those $12 chinese desoldering hotplates (remember to put something heat resistant under it - they come mounted on legs. Thermally conductive metal legs. They'll leave scorch marks on your desk). That thing is one of the most useful $12 pieces of junk I've gotten from china in months. incredible for rework of QFN, and for fixing QFP packages not lined up right - I reflow everything, then all the ones where the QFN package isn't soldered properly go onto the hotplate. You can even swap the chip out, which was previously hopeless even for TQFP - only alternative I had was titanium spatulas, but those make it too easy to lift traces. .

**Reply by SpenceKonde** [reply]
I will note that I can still only compile anything when 1.8.16 is installed as portable, otherwise I get a panic, no major version found when building anything. It's unclear whether that's from a board manager package or a non-board manager package - Ive got 1 board manager package (digispark, for the upload tool so I can upload to ATTinyCore defined boards that use micronucleus bootloader) and a lot of non-board manager ones.

**Comment by ebelouet** [comment]
HI,
I have 1.8.16 arduino and I have nothing on led, where is mistake ?SincerelyEric

    Le lundi 8 novembre 2021 à 08:46:44 UTC+1, Spence Konde (aka Dr. Azzy) ***@***.***> a écrit :  
 
 


I think it's back to the old "assume a new release is no good until proven otherwise" protocol. It's not like they've been putting anything useful in.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
